src/Data/Semigroup/Traversable.hs

```diff
@@ -29,11 +32,34 @@ import Data.Semigroup
 #endif
 import Data.Semigroup.Traversable.Class
 import Data.Functor.Bind.Class
+import GHC.Generics (Rep1, Generic1, to1, from1)
 
 -- | Default implementation of 'foldMap1' given an implementation of 'Traversable1'.
 foldMap1Default :: (Traversable1 f, Semigroup m) => (a -> m) -> f a -> m
 foldMap1Default f = getConst . traverse1 (Const . f)
 
+-- | Generic implementation of 'traverse1'.
+-- 
+-- There are some caveats:
+--
+--   1. It doesn't compile if 't' is an empty constructor
+--   2. It doesn't compile if 't' has some fields that don't mention 'a', for exmaple @data Bar a = MkBar a Int@
+--
+-- @since 5.3.8
+gtraverse1 ::
+  (Traversable1 (Rep1 t), Apply f, Generic1 t) =>
+  (a -> f b) ->
+  t a ->
+  f (t b)
+gtraverse1 f x = to1 <$> traverse1 f (from1 x)
+
+-- | @since 5.3.8
+gsequence1 ::
```

Comment:
Same caveats, and probably the same caveats for the generic `Foldable1` functions too?